### DISCUSSION TITLE
Any known benchmarks with other HTTP servers?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [cesanta/mongoose#1817] [Ask questions, get answers] [answered]
- My goal is: I am looking into implementing a Matrix server and thus became curious about the performance of various web server implementations. The TL;DR is that Matrix only really needs HTTP/1 and TLS/HTTPS for secure message transmission. Has anyone seen or heared about a shootout between several webserver implementations?
- My actions were: I have been looking at h2o, Mongoose, kore and facil.io.
- My expectation was: While taking a few google searches, I hoped to find such a shootout. Found none, unfortunately.
- The result I saw: None, sadly.
- My question is: Above :)

# Environment

- mongoose version: n/a
- Compiler/IDE and SDK: n/a
- Target hardware/board: n/a
- Connectivity chip/module: n/a

**Comment by scaprile** [comment, marked as answer]
Well, given that Mongoose is an embedded library, the amount of requests that can be handled depends on the TCP/IP stack, the microcontroller, the RTOS if any... you should build all candidates on the same target with an equivalent configuration and run _siege_ or something.
What we did here is to benchmark Mongoose over different TCP/IP stacks (including our own), running on the same dev board:
https://github.com/cesanta/mongoose/tree/master/examples/stm32/nucleo-f746zg-baremetal#benchmark
